src/components/AddApplication/AddApplication.js

```diff
@@ -0,0 +1,138 @@
+            setState('finished');
+            loadEntities();
+        })
+        .catch(({ data: { errors: [{ detail }] } }) => {
+            setState('errored');
+            setError(detail);
+            setValues({ application });
+        });
+    };
+
+    if (state !== 'wizard') {
+        return (
+            <Wizard
+                isOpen={ true }
+                onClose={ history.goBack }
+                title={intl.formatMessage({
+                    id: 'sources.addApp',
+                    defaultMessage: 'Add application'
+                })}
+                description={
+                    intl.formatMessage({
+                        id: 'sources.addAppDescription',
+                        defaultMessage: 'You are importing data into this platform'
+                    })
+                }
+                steps={ [{
+                    name: 'Finish',
+                    component: state === 'loading' ? <LoadingStep />
+                        : state === 'finished' ? <FinishedStep
+                            onClose={history.goBack}
+                            successfulMessage={<FormattedMessage
+                                id="sources.successAddApp"
+                                defaultMessage="Your application has been successfully added."
+                            />}
+                            title={<FormattedMessage
+                                id="sources.configurationSuccessful"
+                                defaultMessage="Configuration successful"
+                            />}
+                        /> : <ErroredStep
+                            onClose={history.goBack}
+                            message={
+                                <React.Fragment>
+                                    <FormattedMessage
+                                        id="sources.successAddApp"
+                                        defaultMessage="Your application has not been successfully added:"
+                                    />
+                                    <br />
+                                    <TextContent>
+                                        <Text component={TextVariants.h6}>{ error }</Text>
+                                    </TextContent>
+                                </React.Fragment>
+                            }
+                            title={<FormattedMessage
+                                id="sources.configurationSuccessful"
+                                defaultMessage="Configuration unsuccessful"
+                            />}
+                            onRetry={() => setState('wizard')}
+                        />,
+                    isFinishedStep: true
+                }] } />
+        );
+    }
+
+    return (
+        <SourcesFormRenderer
+            schema={schema}
+            showFormControls={false}
+            onSubmit={filteredAppTypes.length > 0 ? onSubmit : history.goBack}
```

Comment:
I would use `history.push` instead of `goBack`. What if there is no route to come back to?